contracts/PoSWRental.sol

```diff
@@ -0,0 +1,44 @@
+pragma solidity ^0.5.1;
+
+contract PoSWRental{
+    address payable owner_;
+    address payable miner_;  // miner address, should be payble (not be a smart contract)
+    uint256 principal_;
+    uint256 minerRate_;     // in 1/10000
+
+    constructor (address payable owner, uint256 minerRate) public {
+        owner_ = owner;
+        minerRate_ = minerRate;
+        principal_ = address(this).balance;
+    }
+
+    function AddPrincipal() payable external {
+        principal_ = principal_ + msg.value;
+    }
+
+    function WithdrawPrincipal(uint amount) public {
+        require(msg.sender == owner_);
+        require(principal_ >= amount);
+        owner_.transfer(amount);
+        principal_ = principal_ - amount;
+    }
+
+    function GetDivident() public view returns (uint256) {
+        return address(this).balance - principal_;
+    }
+
+    function Payout() public {
+        uint256 amount = GetDivident() * minerRate_ / 10000;
+
+        // Ignore failure if miner is a contract
+        miner_.send(amount);
```

Comment:
comment says miner should not be a contract, but we are not enforcing miner being non-contract (by checking its `code`)

Reply:
I don't we could enforce as the miner address could be contract address but the contract is not created yet (then bypass the codesize == 0 check).  I think there is a similar security issue we discussed before.  As a result, it is the miner's responsibility to ensure the address is a user one.